reconcile/utils/terraform_client.py

```diff
@@ -575,34 +577,74 @@ def cleanup(self):
-            and self._aws_api is not None
-        ):
+        pending_values: list[str] = []
+        if changed_resource_arguments and self._aws_api is not None:
             region_name = get_region_from_availability_zone(before["availability_zone"])
-            response = self._aws_api.describe_rds_db_instance(
-                account_name, resource_name, region_name=region_name
-            )
-            pending_modified_values = response["DBInstances"][0][
+            try:
+                response = self._aws_api.describe_rds_db_instance(
+                    account_name, resource_name, region_name=region_name
+                )
+            except ClientError as e:
+                # The RDS database might have been already removed, or there was
+                # a resource name change, and we no longer use valid references.
+                # There is no need to delay apply.
+                if e.response.get("Error", {}).get("Code") == "DBInstanceNotFound":
+                    logging.debug(f"Resource does not exist: {resource_name}")
+                    return pending_values
+                else:
+                    raise
+            pending_modified_values = response["DBInstances"][0].get(
                 "PendingModifiedValues"
-            ]
-            if (
-                "EngineVersion" in pending_modified_values
-                and pending_modified_values["EngineVersion"]
-                == resource_change["after"]["engine_version"]
-            ):
-                return True
-        return False
+            )
+            # The PendingModifiedValues attribute might not be included as part
+            # of the RDS database instance object. However, we expect it to be
+            # included here, especially since, at this point, Terraform also
+            # claims a pending change for the underlying resource. Thus if we
+            # can't agree on AWS API vs Terraform state, then perhaps changes
+            # have already been applied.
+            if pending_modified_values is None:
+                return pending_values
+            for argument in changed_resource_arguments:
+                # We ignore anything that is not supported.
+                value = supported_values.get(argument)
+                if (
+                    value in pending_modified_values
+                    and cast(dict[str, str], pending_modified_values)[value]
+                    == after[argument]
+                ):
+                    pending_values.append(argument)
+            # A change to the resource on Terraform's side without being reflected
+            # on the AWS' side should not delay the apply, so that a new change to
+            # the underlying resource can be scheduled and added to the set of
+            # pending changes.
+            if set(changed_resource_arguments) - set(pending_values):
+                pending_values = []
+        return sorted([supported_values[v] for v in pending_values])
```

Comment:
@janboll, this changes "instance_class" into "DBInstanceClass" as the log message that uses this information mentions **PendingModifiedValues** (as per how AWS documentation would describe relevant properties in the API) rather than Terraform resource attributes.

Reply:
Right! overlooked that.